src/System.IO.FileSystem/tests/FileInfo/GetSetTimes.cs

```diff
@@ -43,20 +43,16 @@ public FileInfo GetNonZeroMilliseconds()
         public FileInfo GetNonZeroNanoseconds()
         {
             FileInfo fileinfo = new FileInfo(GetTestFilePath());
-            for (int i = 0; i < 20; i++)
-            {
-                fileinfo.Create().Dispose();
-                if (HasNonZeroNanoseconds(fileinfo.LastWriteTime))
-                    break;
+            fileinfo.Create().Dispose();
 
-                // This case should only happen 1/10 times, unless the OS/Filesystem does
-                // not support nanosecond granularity.
-
-                // If it's 1/10, or low granularity, this may help:
-                Thread.Sleep(123);
+            if (!HasNonZeroNanoseconds(fileinfo.LastWriteTime))
```

Comment:
Which OS / FS combination requires this?

Reply:
This is to test that we don't drop nanoseconds from the timestamp while copying on every os\fs